Clarify contract (#8)
term-world/syllabus-template: CODE_OF_CONDUCT.md

# CMPSC 100: Course Contract

## A word from the Mayor's Office

We're building a world. 

That task shouldn't be taken on as a solitary or easy burden. In many ways, world-bulding is always imperfect; we learn more as we _do_, and that process is rarely (i.e. never) linear or "clean." We benefit from being able to be messy and make many mistakes. You may gather from this statement our approach cares about _process_ much more than _product_, though the latter is necessary. Following the various processes of `term-world` closely should yield a successful product. That written, production is a _basic_ expectation of the work we'll complete together. Once the process is over, we need to have a result.

> However, there are manifold tasks involved that are often under- or _not_ counted in traditional grading. That's why community members are evaluated on a _contract basis_, the terms of which are spelled out here. This document specifies your responsibilities and how that work will be rewarded. 

Reviewer: The "under- or _not_ counted" bit is a bit difficult to read, imho.

Author: We'll go with _undercounted_.

## Contract stipulations

### The `Basic Grade`

* All citizens who meet the basic covenants of this contract will earn an individual letter grade of `C` in this course

You may ask why the `Basic Grade` is a `C` and _not a `B` or `B-`_.

Rising to the level of above-average or exceptional work relies on the just that: above-average or exceptional work. Merely completing the assignments maintains a kind of _status quo_. As our interest centers on making a _better_ world, this contract should provide incentives to think beyond mere sufficiency. As you'll see below, work that aspires to higher levels of reward involves doing the work of planning and (better yet) _making_ improvements.

### Improvements

**Opportunities to implement improvements will begin during the `Bodega` assignment.**

Much of the language below will not make complete sense until we're into the collaborative portion of our work. That will come a bit later.

Improvements are small additions to assignments that extend your service in `term-world`. They serve as a way to:

* gain extra practice with course concepts
* demonstrate an above-average level of effort

> Each "improvable" assignment will suggest `10` potential improvements to choose from. Selecting from this list, improvements must be unique (i.e. two students cannot complete the same improvement).

Reviewer: Clarify--two students _in the same group_ cannot complete the same improvement.

#### Improvement `Proposal`s

> Improvements must go through a formal proposal process. To propose an assignment-related world improvment, citizens must use the `Improvement Proposal` template for a given GitHub repository's `Issues` which:

Reviewer 2: ```suggestion
Improvements must go through a formal proposal process. To propose an assignment-related world improvement, citizens must use the `Improvement Proposal` template for a given GitHub repository's `Issues` which:
```

  * responds to the templates questions/prompts
  * is completed by the start of a lab session of a given week
  * must be approved _in writing_ by the Mayor by the end of the lab session
    * the Office of the Mayor will provide suggestions and approval via the `Issue` filed to propose the improvement
  * must either incorporate suggested changes or provide rationale for _not_ incorporating them

> #### Improvment `Implementation`

Reviewer: Improvement

Author: I was thinking they'd have to make it up on the fly with the scenario that they're newscasters and one of them can't see the green screen behind them and the others try to give them hints but it's really hard because they also have to be funny and...

You're right. It needs an `e`.

> Proposing an improvment allows a citizen to _implement_ (i.e. make) that improvement. To implement an improvement:

Reviewer: improvement (that "e" key, huh?)

Author: See above.

   * create a separate `branch` for improvement work, which features an appropriate title
   * make all relevant changes on that `branch`
   * `push` that `branch` to GitHub
   * open a `Pull Request` via the assignment repository
   * link the `Issue` containing the proposal to the created `Pull Request`
     * do this using the `Development` section of the `Issue` (located on the right sidebar)

>    * request review of at least `3` neighorhood members and `1` member of the Office of the Mayor

Reviewer 2: ```suggestion
   * request review of at least `3` neighborhood members and `1` member of the Office of the Mayor
```

|:------------------|:---------------------|:-------------------|:------------------------|
|`A`                | 10                   |8                   |8                        |
|`A-`               | 10                   |8                   |6                        |
|`B+`               | 10                   |6                   |4                        |
|`B`                | 10                   |4                   |4                        |
|`B-`               | 10                   |4                   |2                        |
|`C+`               | 10                   |2                   |1                        |
|`C`                | 10                   |-                   |-                        |

See [below](#grades-lower-than-the-basic-grade) for elaboration on letter grades below the `Basic Grade`.

#### Grades lower than the Basic Grade

There are three (3) situations which penalize evaluation:

1. `Incomplete` assignments (i.e. basic expectations are not fulfilled)
2. `Ignored` assignments (i.e. no evidence of assignment attempt)
3. Plagiarism or other inappropriate/dishonest academic behavior

> Each of these circumstances affects evaluatoin differently.

Reviewer: evaluation

> In severe cases, academic dishonesty may be referred to the supranational Alleghey College Honor Code Committee though these cases are rare (i.e. _almost_ non-existent). For questions of what qualitfies under the provision of "academic dishonesty" see the remainder of this document and the _Compass_.

Reviewer 2: ```suggestion
In severe cases, academic dishonesty may be referred to the supranational Allegheny College Honor Code Committee though these cases are rare (i.e. _almost_ non-existent). For questions of what qualifies under the provision of "academic dishonesty" see the remainder of this document and the _Compass_.
```

#### Bug bounty

To mitigate future issues the Mayor has instituted a "bug bounty." This is a way for you, the citizens, to surface issues with the world. To report a "bug," a citizen must:

* complete a `Bug Report` template on a given repository's `Issues` tab
  
A member of the Office of the Mayor may be in touch to ask about reproducing or observing the bug. Should the report be classified as a "bug," a scaled, appropriate non-letter grade reward will be awarded and announced to citizens at-large.

### The "Rules Rule"

> Convenants in this contract may be changed. This requires a citizen or group of citizens to:

Reviewer 2: ```suggestion
Covenants in this contract may be changed. This requires a citizen or group of citizens to:
```

  * make a successful "Pull Request" on this document _and_
  * lead an open course discussion about the proposed change(s)

Any changes will then be put to a vote.

  * If a vote succeeds with a two-thirds majority (66%) outcome supporting the measure(s) proposed, the change will be made 

The above applies to all portions of the contract _except_ for the `Basic Grade` and this rule ("Rules Rule").

The Mayor reserves the following rights:

  * the right to veto a proposed change if:

>     * the change substantively alters the academic or course community integrity or safety

Reviewer 2: ```suggestion
    * the change substantially alters the academic or course community integrity or safety
```

    * a rule is superseded by College policies or academic standards

>   * the right make unilateral changes that that uphold academic and community integrity or safety

Reviewer: the right _to_ make...

Also only one "that"

* Remind any individual of the orders in which certain commands need to be given
* Watch any individual write code and help guide them, while still allowing room for the individual to think and figure out how to get around obstacles on their own

#### Individual assignments

> While certain assignments should be achieved individually citizens have access to collaborators. However, keep in mind that, like "group" assigments, in pursuit of individual assignments:

Reviewer 2: ```suggestion
While certain assignments should be achieved individually citizens have access to collaborators. However, keep in mind that, like "group" assignments, in pursuit of individual assignments:
```
